server/recurring/amount_test.go

```diff
@@ -0,0 +1,397 @@
+		}
+
+		fmt.Println()
+
+		{
+			elbowIndex := findElbow(amounts)
+			higher := amounts[elbowIndex:]
+			fmt.Println("Above elbow")
+			graph = asciigraph.PlotMany([][]float64{
+				higher,
+			}, asciigraph.Height(35), asciigraph.Width(200))
+			fmt.Println(graph)
+		}
+
+	})
+}
+
+func findElbow(input []float64) int {
+	// normalized_numbers = (numbers - np.min(numbers)) / (np.max(numbers) - np.min(numbers))
+	dup := make([]float64, len(input))
+	copy(dup, input)
+	normalizeShit(dup)
+
+	// graph := asciigraph.PlotMany([][]float64{
+	// 	dup,
+	// }, asciigraph.Height(35), asciigraph.Width(200))
+	// fmt.Println(graph)
+
+	elbowIndex := findElbowPoint(dup)
+	// value := dup[elbowIndex]
+
+	fmt.Println("ELBOW:", elbowIndex, input[elbowIndex], len(input))
+
+	return elbowIndex
+
+}
+
+func normalizeShit(input []float64) {
+	low, high := slices.Min(input), slices.Max(input)
+
+	for i := range input {
+		input[i] = (input[i] - low) / (high - low)
+	}
+}
+
+func findElbowPoint(points []float64) int {
+	distances := calculateDistances(points)
+
+	maxDistance, elbowIndex := 0.0, 0
+	for i, distance := range distances {
+		if distance > maxDistance {
+			maxDistance = distance
+			elbowIndex = i
+		}
+	}
+
+	return elbowIndex
+}
+
+func calculateDistances(points []float64) []float64 {
+	// Line endpoints
+	startPointX, startPointY := 1.0, points[0]
+	endPointX, endPointY := float64(len(points)), points[len(points)-1]
+
+	// Line vector
+	lineVecX, lineVecY := endPointX-startPointX, endPointY-startPointY
+
+	// Slice to store distances
+	distances := make([]float64, len(points))
+
+	// Calculate distance for each point
+	for i, point := range points {
+		pointVecX, pointVecY := float64(i+1)-startPointX, point-startPointY
+		lineLength := math.Sqrt(lineVecX*lineVecX + lineVecY*lineVecY)
+		distance := math.Abs(lineVecX*pointVecY-lineVecY*pointVecX) / lineLength
+		distances[i] = distance
+	}
+
+	return distances
+}
+
+func TestAmountDifferent(t *testing.T) {
+	t.Run("KDE", func(t *testing.T) {
+		items := GetFixtures(t, "amazon_sample_data_1.json")
+		data := make([]float64, len(items))
+		for i, item := range items {
+			data[i] = float64(item.Amount)
+		}
+
+		freq := map[int64]int64{
+			-3:    1,
+			20045: 1,
+			20076: 1,
+			2141:  1,
+			21578: 1,
+			22882: 1,
+			24159: 1,
+			24189: 1,
+			24754: 1,
+			24791: 1,
+			24819: 1,
+			24838: 1,
+			24:    1,
+			7656:  1,
+			8127:  1,
+			8159:  1,
+			8363:  1,
+			8402:  1,
+			8419:  1,
+			9018:  1,
+		}
+		// freq = map[int64]int64{
+		// 	1500: 10,
+		// 	1700: 15,
+		// }
+		fmt.Sprint(freq)
+		// data = make([]float64, 0)
+		// for amount, count := range freq {
+		// 	for i := 0; i < int(count); i++ {
+		// 		data = append(data, float64(amount))
+		// 	}
+		// }
+
+		sort.Float64s(data)
+
+		// Estimate the bandwidth
+		bandwidth := SilvermansRuleOfThumb(data)
+
+		bandwidths := make([]float64, 0)
+		for i := 500; i < 5000; i += 10 {
+			bandwidths = append(bandwidths, float64(i))
+		}
```

Comment:
Should also append the bandwidth from silvermans rule of thumb here too and then sort the array.

We generate several bandwidths starting at $5.00 and then going up by 10 cents, but including the silvermans rule might cause this to be even more accurate since it has no such constraint. OR we might be able to tune our own increment to be 50 cents or even $1 based instead.